services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidationSingleSwFlowSpec.groovy

```diff
@@ -0,0 +1,406 @@
+        northbound.deleteMeter(sw.dpId, meterIds[0])
+        northbound.deleteMeter(sw.dpId, meterIds[1])
+
+        then: "Meters info/rules are moved into the 'missing' section"
+        Long burstSize = switchHelper.getExpectedBurst(sw.dpId, flow.maximumBandwidth)
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        def createdCookies = сookiesWithMeter(sw.dpId)
+        switchValidateInfo.rules.missing.collect { it }.containsAll(createdCookies)
+        switchValidateInfo.meters.missing*.meterId.containsAll(meterIds)
+        switchValidateInfo.meters.missing*.cookie.containsAll(createdCookies)
+        switchValidateInfo.meters.missing.each {
+            assert it.rate == flow.maximumBandwidth
+            assert it.flowId == flow.id
+            assert flagsOnSwitch.containsAll(it.flags)
+            burstSizeIsCorrect(burstSize, it.burstSize)
+        }
+
+        and: "The rest fields are empty"
+        ruleSectionsAreEmpty(switchValidateInfo, ["proper", "excess"])
+        meterSectionsAreEmpty(switchValidateInfo, ["misconfigured", "proper", "excess"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that switch validate info is also deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            def switchValidateInfoAfterDelete = northbound.switchValidate(sw.dpId)
+            ruleSectionsAreEmpty(switchValidateInfoAfterDelete)
+            meterSectionsAreEmpty(switchValidateInfoAfterDelete)
+        }
+
+        where:
+        switchType   | switches               | flagsOnSwitch
+        "Centec"     | getCentecSwitches()    | ["KBPS", "BURST", "STATS"]
+        "non-Centec" | getNonCentecSwitches() | ["KBPS", "BURST", "STATS"]
+    }
+
+    @Unroll
+    def "Switch validation is able to detect meter info into the 'excess' section on a #switchType switch"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        setup: "Select a #switchType switch and retrieve default meters"
+        def sw = switches.first()
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def metersIds = createdMeterIds(sw.dpId)
+        Long burstSize = switchHelper.getExpectedBurst(sw.dpId, flow.maximumBandwidth)
+
+        and: "Update meterId for created flow directly via db"
+        def newMeterId = 100
+        database.updateFlowMeterId(flow.id, newMeterId)
+
+        then: "Meters info are moved into the 'excess' section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.meters.excess.collect { it.meterId }.containsAll(metersIds)
+        switchValidateInfo.meters.excess.each {
+            assert it.rate == flow.maximumBandwidth
+            assert flagsOnSwitch.containsAll(it.flags)
+            burstSizeIsCorrect(burstSize, it.burstSize)
+        }
+
+        and: "Meters info is also stored in the 'missing' section"
```

Comment:
>         then: "Meters info are moved into the 'excess' section"
>        and: "Meters info is also stored in the 'missing' section"
from this context is seems that you have the same meters in both sections, which is not what happens

Reply:
fixed